server/src/testflinger/api/v1.py

```diff
@@ -640,8 +640,19 @@ def images_post():
 @v1.output(schemas.AgentOut(many=True))
 def agents_get_all():
     """Get all agent data."""
-    agents = database.mongo.db.agents.find({}, {"_id": False, "log": False})
-    return jsonify(list(agents))
+    agents = database.get_agents()
+    restricted_queues = database.get_restricted_queues()
+    restricted_queues_owners = database.get_restricted_queues_owners()
+
+    for agent in agents:
+        agent["restricted_to"] = {
+            queue: restricted_queues_owners.get(queue, [])
+            if queue in restricted_queues
+            else []
+            for queue in agent.get("queues", [])
+        }
```

Comment:
Before merging I was checking which was the output of the endpoint and I realize it returns an empty list for each of the queues in the agent:

```
   "restricted_to": {
      "test_queue1": []'
    }
```

I was thinking maybe it makes more sense to return an empty object if is no restriction at all?
`"restricted_to": {}`

```suggestion
    for agent in agents:
        agent["restricted_to"] = {
            queue: restricted_queues_owners[queue]
            for queue in agent.get("queues", [])
            if queue in restricted_queues
            and restricted_queues_owners.get(queue)
        }
```
Since this might affect the design on the UI PR that you are also working on, please let me know what do you think. Basically just wanted to know if the current return is the one that you are expecting